File: Prid1920-g03/Models/Comment.cs
using System;
using System.Linq;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics;
using Microsoft.EntityFrameworkCore;

namespace Prid1920_g03.Models
{

    public class Comment
    {
        [Key]
        public int Id { get; set; }

        [Required(ErrorMessage = "Required")]
        public string Body { get; set; }

        [Required(ErrorMessage = "Required")]
        public DateTime Timestamp { get; set; }

        public int PostId { get; set; }
        public int AuthorId { get; set; }
        public virtual User User {get; set;}
        public virtual Post Post { get; set; }
    }
}

File: Prid1920-g03/ClientApp/src/app/models/comment.ts

  export class Comment {
    id: number;
    body: string;
    timestamp: string; 
    authorId: number;
    postId: number;
    commentUser: any;

      constructor(data: any) {
        if(data){
          this.id = data.id;
          this.body = data.body;
          this.timestamp = data.timestamp;
          this.authorId = data.authorId;
          this.postId = data.postId;
          this.commentUser = data.commentUser;
        }
      }
  }
    


